cli/bin/grainrun.js

```diff
@@ -31,7 +31,18 @@ const importObject = { wasi_snapshot_preview1: wasi.wasiImport };
 
 async function run(filename) {
   try {
-    const wasm = await WebAssembly.compile(await readFile(filename));
+    const wasm = await WebAssembly.compile(await readFile(filename)).catch(
+      (err) => {
+        if (filename.endsWith(".gr")) {
+          console.log(
+            `Grain run is for compiled wasm files, to compile a grain file use grain ${filename}`
+          );
+        } else {
+          console.log("Invalid wasm file detected");
+        }
+        process.exit();
+      }
+    );
```

Comment:
I think I want to use try/catch with the awaits instead of `.catch` so we need to restructure this a big.

Reply:
I made that change if we want as well we could make a prettier error for a file that doesn't exist. and we can try and make runtime errors a little prettier might be a job for a different pr though.

Reply:
I've updated further